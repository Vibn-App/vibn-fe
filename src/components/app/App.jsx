import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import UserProfile from '../displays/UserProfile';
import Signin from '../displays/Signin';
import UserList from '../displays/UserList';
import UserDetail from '../displays/UserDetail';
import Conversations from '../displays/Conversations';
import LoggedIn from '../displays/LoggedIn';
import About from '../displays/About';
import CreateConvo from '../displays/CreateConvo';
import ConvoDetail from '../displays/ConvoDetail';

export default function App() {
  return (
    <div>
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={Signin} />
          <Route exact path="/user/:id/:access_token/:refresh_token" component={LoggedIn} />
          <Route path="/user/:id/dash">
            <UserProfile />
          </Route>
          <Route exact path="/users">
            <UserList />
          </Route>
          <Route exact path="/users/:id">
            <UserDetail />
          </Route>
          <Route exact path="/user/:id/conversations">
            <Conversations />
          </Route>
          <Route exact path="/about">
            <About />
          </Route>
          <Route exact path="/user/create/convo/:id">
            <CreateConvo />
          </Route>
          <Route exact path="/user/convo/detail/:id">
            <ConvoDetail />
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
}
