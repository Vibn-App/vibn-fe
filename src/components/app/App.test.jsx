import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App', () => {
  it('renders app', () => {
    render(<App />);

    const app = screen.getByText('vibn');
    expect(app).toMatchSnapshot();
  });
});
