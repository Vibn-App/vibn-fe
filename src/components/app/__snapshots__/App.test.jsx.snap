// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app 1`] = `
<h2
  class="vibn"
>
  vibn
</h2>
`;
